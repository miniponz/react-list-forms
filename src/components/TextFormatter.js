import React, { PureComponent } from 'react';

export default class TextFormatter extends PureComponent {
  state = {
    text: 'type here',
    color: 'black'
  }

  handleSubmit = event => {
    event.preventDefault();
  }

  handleTextChange = ({ target }) => {
    this.setState({ text: target.value });
  }

  handleColorChange = ({ target }) => {
    this.setState({ color: target.value });
  }

  render() {
    const {
      text,
      color
    } = this.state;

    return (
      <>
      <form onSubmit={this.handleSubmit}>
        <input type="color" onChange={this.handleColorChange}/>
        <input name="text" value={text} onChange={this.handleTextChange} type="text" />
        <button>Enter</button>
      </form>
      <div>
        <p style={{ color: color }}>{text}</p>
      </div>
      </>
    );
  }
}

{/* <select onChange={this.handleColorChange}>
<option value={'black'}>black</option>
<option value={'blue'}>blue</option>
<option value={'red'}>red</option>
</select> */}
