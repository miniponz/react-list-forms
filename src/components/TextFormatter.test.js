import React from 'react';
import { shallow } from 'enzyme';
import TextFormatter from './TextFormatter';

describe('text formatter', () => {
  it('renders text', () => {
    const wrapper = shallow(<TextFormatter />);
    expect(wrapper).toMatchSnapshot();
  });
});
