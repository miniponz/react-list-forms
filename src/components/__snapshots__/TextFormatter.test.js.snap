// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colors list renders color list 1`] = `
<Fragment>
  <form>
    <select>
      <option>
        not working
      </option>
      <option>
        still not working
      </option>
    </select>
    <input
      name="text"
      value=""
    />
    <button>
      Enter Text
    </button>
  </form>
  <div>
    <p
      style={
        Object {
          "color": "",
        }
      }
    />
  </div>
</Fragment>
`;

exports[`text formatter renders color list 1`] = `
<Fragment>
  <form>
    <select>
      <option>
        not working
      </option>
      <option>
        still not working
      </option>
    </select>
    <input
      name="text"
      value=""
    />
    <button>
      Enter Text
    </button>
  </form>
  <div>
    <p
      style={
        Object {
          "color": "",
        }
      }
    />
  </div>
</Fragment>
`;

exports[`text formatter renders text 1`] = `
<Fragment>
  <form>
    <select>
      <option>
        not working
      </option>
      <option>
        still not working
      </option>
    </select>
    <input
      name="text"
      value=""
    />
    <button>
      Enter Text
    </button>
  </form>
  <div>
    <p
      style={
        Object {
          "color": "",
        }
      }
    />
  </div>
</Fragment>
`;

exports[`text list renders color list 1`] = `
<Fragment>
  <form>
    <select>
      <option>
        not working
      </option>
      <option>
        still not working
      </option>
    </select>
    <input
      name="text"
      value=""
    />
    <button>
      Enter Text
    </button>
  </form>
  <div>
    <p
      style={
        Object {
          "color": "",
        }
      }
    />
  </div>
</Fragment>
`;
